src/main/java/io/github/thebusybiscuit/slimefun4/utils/itemstack/ColoredFireworkStar.java

```diff
@@ -1,51 +1,37 @@
 package io.github.thebusybiscuit.slimefun4.utils.itemstack;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import javax.annotation.ParametersAreNonnullByDefault;
-
+import io.github.bakedlibs.dough.items.ItemStackEditor;
+import io.github.thebusybiscuit.slimefun4.utils.compatibility.VersionedItemFlag;
 import org.bukkit.Color;
 import org.bukkit.FireworkEffect;
 import org.bukkit.FireworkEffect.Type;
 import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.FireworkEffectMeta;
 
-import io.github.bakedlibs.dough.common.ChatColors;
-import io.github.bakedlibs.dough.items.CustomItemStack;
-import io.github.thebusybiscuit.slimefun4.utils.compatibility.VersionedItemFlag;
+import javax.annotation.ParametersAreNonnullByDefault;
 
 /**
  * This simple {@link ItemStack} implementation allows us to obtain
  * a colored {@code Material.FIREWORK_STAR} {@link ItemStack} quickly.
- * 
+ *
  * @author TheBusyBiscuit
  *
  */
-public class ColoredFireworkStar extends CustomItemStack {
+public class ColoredFireworkStar {
 
     @ParametersAreNonnullByDefault
-    public ColoredFireworkStar(Color color, String name, String... lore) {
-        super(Material.FIREWORK_STAR, im -> {
-            if (name != null) {
-                im.setDisplayName(ChatColors.color(name));
-            }
-
-            ((FireworkEffectMeta) im).setEffect(FireworkEffect.builder().with(Type.BURST).withColor(color).build());
-
-            if (lore.length > 0) {
-                List<String> lines = new ArrayList<>();
-
-                for (String line : lore) {
-                    lines.add(ChatColors.color(line));
-                }
-
-                im.setLore(lines);
-            }
-
-            im.addItemFlags(VersionedItemFlag.HIDE_ADDITIONAL_TOOLTIP);
-        });
+    public static ItemStack create(Color color, String name, String... lore) {
+        return new ItemStackEditor(Material.FIREWORK_STAR)
+                .setDisplayName(name)
+                .setLore(lore)
+                .addFlags(VersionedItemFlag.HIDE_ADDITIONAL_TOOLTIP)
+                .withMetaConsumer(meta -> {
```

Comment:
Yeah I mean, that one's a bit more annoying to implement. At that point it's like, why not just call paper's `ItemStack#editMeta(Class<T>, Consumer<T>)` after the `#create` call for something so specific.

Reply:
Because having to call it afterwards is annoying, and instead I’d just do what you’re doing here currently even if it’s bulkier. I also think it would be used often enough with the changes to itemstack components now that everything can be used with damage able, etc. I don’t see why it’d be too hard to implement though, just get meta, class#isinstance, consumer.accept(class.cast()).

Reply:
I'd have to store the type. ItemStackEditor does not ever modify the internal item stack, it only stores the transformations as a Consumer. Also, it would make it difficult to determine the order of the transformations since now there is no singular type/consumer.

Reply:
You could do something like: 
![image](https://github.com/user-attachments/assets/8154b078-4650-4d2e-8197-30987aad764c)
I don't think it would have the problems you described

Reply:
Yeah, sure. I will note that there is a side effect in that once a consumer is declared, even if it does nothing, an ItemMeta instance is created and set upon the ItemStack. So actually ItemStack#editMeta has slightly different semantics than what you described/what ItemStackEditor does for these casea.

Reply:
I never said it was supposed to be identical to ItemStack#editMeta 😅 all I was saying was this would be a useful thing for the editor to have and this is a use case for it